File: FiscalFlow/FiscalFlow.Domain/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using FiscalFlow.Domain.Core.Abstractions;
using FiscalFlow.Domain.Core.Primitives;
using FiscalFlow.Domain.Enums;
using NodaMoney;
using MyCurrency = FiscalFlow.Domain.Enums.MyCurrency;

namespace FiscalFlow.Domain.Entities;

public class Transaction : BaseEntity
{
    [NotMapped] public Money Value => new(MoneyValue, MoneyCurrency.ToString());
    [JsonIgnore] [Column("Value")] public decimal MoneyValue { get; set; }
    [JsonIgnore] [Column("Currency")] public MyCurrency MoneyCurrency { get; set; }

    [Column("AccountValueBeforeTransaction")]
    public decimal AccountValueBefore { get; set; }

    [Column("AccountValueAfterTransaction")]
    public decimal AccountValueAfter { get; set; }

    public double? Latitude { get; set; }
    public double? Longitude { get; set; }

    [MaxLength(250)] public string Description { get; set; } = string.Empty;

    [MaxLength(100)] public string Payee { get; set; } = string.Empty;
    public TransactionType Type { get; set; }
    public Category Category { get; set; }

    public Guid AccountId { get; set; }
    public virtual Account Account { get; set; } = null!;
}

File: FiscalFlow/fiscalflow.client/src/app/shared/models/transaction/transaction.ts
import { CurrencyEnum } from '../account/enums/currencyEnum';
import { TransactionType } from './enums/transactionType';
import { Category } from './enums/category';

export interface Transaction {
  id: string;
  value: number;
  currency: CurrencyEnum;
  accountValueBefore: number;
  accountValueAfter: number;
  description: string;
  payee: string;
  type: TransactionType;
  imageUrl: string | null;
  longitude: number | null;
  latitude: number | null;
  reccurencePeriod: number | null;
  category: Category;
  createdOnUtc: Date;
  accountId: string;
}
